Azure/azure-cli
# {Network} `az network application-gateway private-link`: Migrate commands to codegen v2

**Related command**

az network application-gateway private-link

**Description**

Total: 10 commands

**Testing Guide**

Evidences for simplifying the logic of `private-link add`:
1. When we input a non-existing subnet ID, service will raise:
> Code: InvalidResourceReference
Message: Resource /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/ethan-rg/providers/Microsoft.Network/virtualNetworks/ag-13Vnet/subnets/default2 referenced by resource /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/ethan-rg/providers/Microsoft.Network/applicationGateways/ag-13 was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.

Hence, when there is no subnet like that, we can just create a new subnet to simplify the logic.

2. When we create a new subnet with an existing prefix, service will raise:
> Code: NetcfgInvalidSubnet 
Message: Subnet 'default2' is not valid in virtual network 'ag-13Vnet'.

Hence, we don't need to put such error handling logic in the codes.

**History Notes**


[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.
